src/System.Runtime.Intrinsics.X86/src/System/Runtime/Intrinsics/X86/Sse42.cs

```diff
@@ -0,0 +1,262 @@
+// =+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+//
+// SSE42.cs
+//
+// A class that implements intrinsic functions to provide access to Intel SSE4.1 instructions.
+//
+// =+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+
+using System;
+using System.Runtime.Intrinsics;
+
+namespace System.Runtime.Intrinsics.X86
+{
+    /// <summary>
+    /// Provides access to Intel SSE4.2 hardware instructions via intrinsics
+    /// </summary>
+    /// <para>
+    /// SSE42 class provides access to 128-bit SSE4.2 SIMD instructions
+    /// </para>
+
+    public static class Sse42
+    {
+        public static bool IsSupported { get; }
+
+        // int _mm_cmpistra (__m128i a, __m128i b, const int imm8)
+        // int _mm_cmpistrc (__m128i a, __m128i b, const int imm8)
+        // int _mm_cmpistro (__m128i a, __m128i b, const int imm8)
+        // int _mm_cmpistrs (__m128i a, __m128i b, const int imm8)
+        // int _mm_cmpistrz (__m128i a, __m128i b, const int imm8)
+        public static bool CompareImplicitLength(Vector128<sbyte> left, Vector128<sbyte> right, ResultsFlag flag, StringComparisonMode mode)
+        {
+            throw new NotImplementedException(); 
+        }
+
+        // int _mm_cmpistra (__m128i a, __m128i b, const int imm8)
+        // int _mm_cmpistrc (__m128i a, __m128i b, const int imm8)
+        // int _mm_cmpistro (__m128i a, __m128i b, const int imm8)
+        // int _mm_cmpistrs (__m128i a, __m128i b, const int imm8)
+        // int _mm_cmpistrz (__m128i a, __m128i b, const int imm8)
+        public static bool CompareImplicitLength(Vector128<byte> left, Vector128<byte> right, ResultsFlag flag, StringComparisonMode mode)
+        {
+            throw new NotImplementedException(); 
+        }
```

Comment:
Nit: the formatting is different in some cases, e.g. everything on one line versus split across multiple lines... is there a reason for when you've used one or the other?